osbenchmark/utils/dataset.py

```diff
@@ -144,6 +148,115 @@ def parse_context(context: Context) -> str:
         raise Exception("Unsupported context")
 
 
+class ParquetDataSet(DataSet):
+    """ Column-oriented data storage format designed for efficient data storage and retrieval
+    """
+
+    FORMAT_NAME = "parquet"
+    DEFAULT_BATCH_SIZE = 500
+
+    def __init__(self, dataset_path: str, column: str):
+        if not dataset_path:
+            raise ConfigurationError("dataset path cannot be None or empty")
+        if not column:
+            raise ConfigurationError("dataset column name cannot be None or empty")
+        self.dataset_path = dataset_path
+        self.file = None
+        self.column = column
+        self.current = self.BEGINNING
+        self.batch_iterator = None
+        self.prev_batch = []
+        self.file_batch_start = 0
+        self.file_batch_end = 0
+
+    def get_batch(self):
+        if self.prev_batch:
+            return self.prev_batch
+        batch = next(self.batch_iterator)
+        self.file_batch_start = self.file_batch_end
+        self.file_batch_end = self.file_batch_start + len(batch)
+        return batch
+
+    def _extract_vectors(self, start_offset, end_offset):
+        """
+        Since Parquet files are huge, it is not advisable to load complete file in memory.
+        Hence, we will iterate as a batch and extract vectors from one or multiple batches
+        based on start and end offset.
+
+        @param start_offset: start offset to extract vectors from parquet file
+        @param end_offset: start offset to extract vectors from parquet file
+        @return: array of vectors
+        """
+        vectors = []
+        while True:
+            batch = self.get_batch()
+            if not batch:
+                break
+            batch_length = len(batch)
+            if start_offset > self.file_batch_end:
```

Comment:
Why do we need this? Cant we seek?

Reply:
I couldn't find a way to seek parquet file. Hence, i have to use batch to perform in-memory seek.